nix/nim-status/build.nix

```diff
@@ -0,0 +1,317 @@
+{ pkgs, stdenv, lib, fetchFromGitHub
+# Dependencies
+, xcodeWrapper
+, srcRaw
+, writeScript
+, openssl
+, pcre
+, writeTextFile
+, androidPkgs
+, git 
+, platform ? "android"
+, arch ? "386"
+, api ? "29" }:
+
+let
+  osId = builtins.elemAt (builtins.split "\-" stdenv.hostPlatform.system) 2;
+  osArch = builtins.elemAt (builtins.split "\-" stdenv.hostPlatform.system) 0;
+
+
+  stdioPatch = ./stdio.patch;
+  ANDROID_NDK_HOME = "${androidPkgs}/ndk-bundle";
+
+  
+  targetArchMap = rec {
+    "386" = "i686";
+    "arm" = "arm";
+    "arm64" = "aarch64";
+  };
+  ldArchMap = {
+    "386" = "x86";
+    "arm" = "arm";
+    "arm64" = "arm64";
+  };
+
+  ldDirMap = {
+    "386" = "x86";
+    "arm" = "armeabi-v7a"; 
+    "arm64" = "arm64-v8a";
+  };
+
+  # Shorthands for the built phase
+  targetArch = lib.getAttr arch targetArchMap;
+  ldArch = lib.getAttr arch ldArchMap;
+  androidTarget = targetArch + "-linux-" + platform;
+
+  # Arg arch -> Nim arch
+  nimCpuMap = {
+    "386" = "i386";
+    "x86_64" = "amd64"; 
+    "arm" = "arm"; 
+    "arm64" = "arm64";
+  };
+
+  iosArchMap = {
+    "386" = "x86_64";
+    "arm" = "armv7";
+    "arm64" = "arm64";
+  };
+
+  nimCpu = if platform == "ios" && arch == "386"
+    then "amd64" else "${lib.getAttr arch nimCpuMap}";
+  nimPlatform = "${(if platform == "ios" then "ios" else "android")}";
+
+  iosSdk = if arch == "386" then "iphonesimulator" else "iphoneos";
+  iosArch = lib.getAttr arch iosArchMap;
+
+  isAndroid = lib.hasPrefix "android" platform;
+  isIOS = platform == "ios";
+
+
+  buildType = lib.getConfig "build-type" "";
+  nimOverride = ./nim_status.nim;
+  applyNimOverride = 
+    if (lib.elem buildType ["release" "nightly"]) then
+    "cp ${nimOverride} src/nim_status/c/nim_status.nim" else "";
+
+  routeHeader = builtins.readFile ./route.h;
+
+  androidIncludes = stdenv.mkDerivation {
+    name = "nim-status-android-includes";
+    buildInputs = [ pkgs.coreutils pkgs.gnused ];
+    builder = writeScript "nim-android-includes.sh"
+    ''
+      export PATH=${pkgs.coreutils}/bin:${pkgs.patch}/bin
+      mkdir $out
+      cd $out
+      cp ${ANDROID_NDK_HOME}/sysroot/usr/include/stdio.h .
+      patch -u stdio.h -i ${stdioPatch}
+    '';
+  };
+  iosIncludes = stdenv.mkDerivation {
+    name = "nim-status-ios-includes";
+    buildInputs = [ pkgs.coreutils ];
+    builder = writeScript "nim-ios-includes.sh"
+    ''
+      export PATH=${pkgs.coreutils}/bin
+      mkdir $out
+      cd $out
+      mkdir net
+      echo "${routeHeader}" > net/route.h
```

Comment:
This is awful. Just use `writeTextDir "net/route.h" ./route.h` instead of `builtins.readFile ./route.h` and get rid of this whole `iosIncludes` thing.

Same goes for